src/Adapter/Local.php

```diff
@@ -63,11 +65,19 @@ public function delete($path)
     {
         $location = $this->applyPathPrefix($path);
 
+        if (!file_exists($location)) {
+            throw new FileNotFoundException($location);
+        }
+
         if (!is_writable($location)) {
-            return false;
+            throw new IOException('File is not writable', $location);
         }
 
-        return unlink($location);
+        try {
+            return Thrower::call('unlink', $location);
+        } catch (\ErrorException $ex) {
+            throw new FileNotFoundException($location, $ex);
+        }
```

Comment:
I don't think we ever pass the actual path, `$location`, to the exception, but rather the abstraction path, `$path`.

I also can't help but think we should check the `$ex` message to see if it contains "No such file or directory" before throwing a `FileNotFoundException`. The else case could be generic `IOException`